Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Effectiveness.mo

```diff
@@ -0,0 +1,165 @@
+  Modelica.Blocks.Interfaces.RealInput y(final unit="1") "Wheel speed ratio"
+    annotation (Placement(transformation(extent={{-140,60},{-100,100}})));
+  Modelica.Blocks.Interfaces.RealOutput epsS(final unit="1")
+    "Sensible heat exchanger effectiveness"
+    annotation (Placement(transformation(extent={{100,30},{120,50}})));
+  Modelica.Blocks.Interfaces.RealOutput epsL(final unit="1")
+    "Latent heat exchanger effectivenessr"
+    annotation (Placement(transformation(extent={{100,-50},{120,-30}})));
+
+protected
+   Real vRat
+   "Ratio of the average operating volumetric air flow rate to the nominal supply air flow rate";
+
+algorithm
+  // calculate effectiveness
+  if TSup > TExh then
+    // cooling mode
+    epsS :=y*(epsS_cool_partload + (epsS_cool_nominal - epsS_cool_partload)*(
+      vRat-0.75)/0.25);
+    epsL :=y*(epsL_cool_partload + (epsL_cool_nominal - epsL_cool_partload)*(
+      vRat-0.75)/0.25);
+  else
+    // heating mode
+    epsS :=y*(epsS_heat_partload + (epsS_heat_nominal - epsS_heat_partload)*(
+      vRat-0.75)/0.25);
+    epsL :=y*(epsL_heat_partload + (epsL_heat_nominal - epsL_heat_partload)*(
+      vRat-0.75)/0.25);
+  end if;
+  epsS := Buildings.Utilities.Math.Functions.smoothMax(
+       x1 = 0.01,
+       x2 = epsS,
+       deltaX = 1E-7);
+  epsS := Buildings.Utilities.Math.Functions.smoothMin(
+       x1 = 0.99,
+       x2 = epsS,
+       deltaX = 1E-7);
+
+  epsL := Buildings.Utilities.Math.Functions.smoothMax(
+       x1 = 0.01,
+       x2 = epsL,
+       deltaX = 1E-7);
+  epsL := Buildings.Utilities.Math.Functions.smoothMin(
+       x1 = 0.99,
+       x2 = epsL,
+       deltaX = 1E-7);
+
+equation
+  // check if the extrapolation goes too far
+  assert(v_flow_Sup - 2*v_flow_Exh < 0 or v_flow_Exh - 2*v_flow_Sup < 0,
+    "Unbalanced air flow ratio",
+    level=AssertionLevel.warning);
+  // calculate the average volumetric air flow and flow rate ratio
+  vRat =  (v_flow_Sup + v_flow_Exh)/2/v_flow_sup_nominal;
+  // check if the extrapolation goes too far
+  assert(vRat > 0.5 and vRat < 1.3,
+    "Operatiing flow rate outside full accuracy range",
+    level=AssertionLevel.warning);
+  annotation (Icon(coordinateSystem(preserveAspectRatio=false), graphics={Text(
+          extent={{-54,28},{50,-40}},
+          textColor={28,108,200},
+          textString="eps")}), Diagram(
+        coordinateSystem(preserveAspectRatio=false)),
+    defaultComponentName="EffCal",
+Documentation(info="<html>
+<p>
+This block calculates the sensible and latent effectiveness of the heat exchanger
+for heating and cooling conditions at different air flow rates of the supply
+air stream and the exhaust air stream.
+</p>
+<p>
+It first calculates the average volumetric air flow rate through the heat exchanger by:
+</p>
+<pre>
+  v_ave = (v_sup + v_exh)/2,
+  vRat = v_ave/v_sup_nom,
+</pre>
+<p>
+where <code>v_ave</code> is the average volumetric air flow rate,
```

Comment:
See how we otherwise refer to volume or mass flow rates. (Moreover, if you were to use mass flow rate, then the sensor would be simpler, and the result would in my opinion not change much if at all).